circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.test.cpp

```diff
@@ -1092,4 +1100,110 @@ TEST_F(native_private_kernel_inner_tests, native_max_read_requests_one_transient
+        read_request_membership_witnesses{};
+
+    new_commitments[0] = fr(1285);
+    new_commitments[1] = fr(1283);
+    new_commitments[2] = fr(1282);
+    new_commitments[3] = fr(1284);
+
+
+    read_requests[0] = fr(1283);
+    read_requests[1] = fr(1284);
+    read_request_membership_witnesses[0].leaf_index = fr(-1);
+    read_request_membership_witnesses[1].leaf_index = fr(-1);
+
+
+    private_inputs.previous_kernel.public_inputs.end.new_commitments = new_commitments;
+    private_inputs.private_call.call_stack_item.public_inputs.read_requests = read_requests;
+    private_inputs.private_call.read_request_membership_witnesses = read_request_membership_witnesses;
+
+    DummyComposer composer =
+        DummyComposer("native_private_kernel_ordering_tests__native_read_requests_choping_commitment_works");
+    auto const& public_inputs = native_private_kernel_circuit_ordering(composer, private_inputs);
+
+    auto failure = composer.get_first_failure();
+    if (failure.code != CircuitErrorCode::NO_ERROR) {
+        info("failure: ", failure);
+    }
+    ASSERT_FALSE(composer.failed());
+    ASSERT_TRUE(array_length(public_inputs.end.new_commitments) == 2);
+    ASSERT_TRUE(public_inputs.end.new_commitments[0] == fr(1285));
+    ASSERT_TRUE(public_inputs.end.new_commitments[1] == fr(1282));
+}
+
+TEST_F(native_private_kernel_ordering_tests, native_read_request_unknown_fails)
+{
+    auto private_inputs = do_private_call_get_kernel_inputs_inner(false, deposit, standard_test_args());
+
+    auto new_commitments = zero_array<fr, KERNEL_NEW_COMMITMENTS_LENGTH>();
+    auto read_requests = zero_array<fr, READ_REQUESTS_LENGTH>();
+    std::array<MembershipWitness<NT, PRIVATE_DATA_TREE_HEIGHT>, READ_REQUESTS_LENGTH>
+        read_request_membership_witnesses{};
+
+    new_commitments[0] = fr(1285);
+    new_commitments[1] = fr(1283);
+    new_commitments[2] = fr(1282);
+    new_commitments[3] = fr(1284);
+
+
+    read_requests[0] = fr(1284);
+    read_requests[1] = fr(1282);
+    read_requests[2] = fr(1283);
+    read_requests[3] = fr(1286);
+    read_request_membership_witnesses[0].leaf_index = fr(-1);
+    read_request_membership_witnesses[1].leaf_index = fr(-1);
+    read_request_membership_witnesses[2].leaf_index = fr(-1);
+    read_request_membership_witnesses[3].leaf_index = fr(-1);
+
+    private_inputs.previous_kernel.public_inputs.end.new_commitments = new_commitments;
+    private_inputs.private_call.call_stack_item.public_inputs.read_requests = read_requests;
+    private_inputs.private_call.read_request_membership_witnesses = read_request_membership_witnesses;
+
+    DummyComposer composer = DummyComposer("native_private_kernel_ordering_tests__native_read_request_unknown_fails");
+    auto const& public_inputs = native_private_kernel_circuit_ordering(composer, private_inputs);
+
+    auto failure = composer.get_first_failure();
+    if (failure.code != CircuitErrorCode::PRIVATE_KERNEL__TRANSIENT_READ_REQUEST_NO_MATCH) {
```

Comment:
I think this `if` condition can be removed. Instead we can add:
`ASSERT_EQ(failure.message, <expected_error_message>)`

Reply:
Addressed!